pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -13,65 +14,83 @@ import (
 	survey "gopkg.in/AlecAivazis/survey.v1"
 )
 
+// Zone represents an AWS route53 DNS Zone
+type Zone struct {
+	ID      string
+	DNSName string
+}
+
 // IsForbidden returns true if and only if the input error is an HTTP
 // 403 error from the AWS API.
 func IsForbidden(err error) bool {
 	requestError, ok := err.(awserr.RequestFailure)
 	return ok && requestError.StatusCode() == http.StatusForbidden
 }
 
-// GetBaseDomain returns a base domain chosen from among the account's
-// public routes.
-func GetBaseDomain() (string, error) {
+// GetBaseDomain returns a base domain zone chosen from among the
+// account's public routes. The zone struct contains the zone ID
+// for the chosen base domain, to distinguish between multiple
+// public zones that share the same base domain.
+func GetBaseDomain() (zone *Zone, err error) {
 	session, err := GetSession()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	logrus.Debugf("listing AWS hosted zones")
 	client := route53.New(session)
-	publicZoneMap := map[string]struct{}{}
-	exists := struct{}{}
+	publicZoneMap := map[string][]string{}
 	if err := client.ListHostedZonesPages(
 		&route53.ListHostedZonesInput{},
 		func(resp *route53.ListHostedZonesOutput, lastPage bool) (shouldContinue bool) {
 			for _, zone := range resp.HostedZones {
 				if zone.Config != nil && !aws.BoolValue(zone.Config.PrivateZone) {
-					publicZoneMap[strings.TrimSuffix(*zone.Name, ".")] = exists
+					zoneName := strings.TrimSuffix(aws.StringValue(zone.Name), ".")
+					publicZoneID := strings.TrimPrefix(aws.StringValue(zone.Id), "/hostedzone/")
+					publicZoneMap[zoneName] = append(publicZoneMap[zoneName], publicZoneID)
 				}
 			}
 			return !lastPage
 		},
 	); err != nil {
-		return "", errors.Wrap(err, "list hosted zones")
+		return nil, errors.Wrap(err, "list hosted zones")
 	}
 
 	publicZones := make([]string, 0, len(publicZoneMap))
 	for name := range publicZoneMap {
-		publicZones = append(publicZones, name)
+		for _, id := range publicZoneMap[name] {
```

Comment:
nit: I think it would be more idiomatic/easier to read:

for name, ids := range publicZonesMap {
		for _, id := range ids {